torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -29,6 +30,40 @@
 )
 from torch.distributed.utils import _to_kwargs
 
+RESHARD_AFTER_FORWARD_STRATEGIES = {
+    HandleShardingStrategy.FULL_SHARD,
+    HandleShardingStrategy.HYBRID_SHARD,
+}
+
+@no_type_check
+def _validate_hybrid_shard_setup(state: _FSDPState, fsdp_module: nn.Module):
```

Comment:
Can we skip this check altogether for composable code path?

In that case, should we change the first argument to `fsdp_root` or something?

Reply:
curious why we need to skip this check for the composable path?

Reply:
The composable path does not support manual wrapping, so there should be no way for the process groups to be inconsistent. In other words, there should be no need to validate the setup. Given this, I recommend writing this method with function argument names that reflect the fact that it is only meant for the non-composable code path.